### DISCUSSION TITLE
stack() function do nothing in dtrace

### DISCUSSION THREAD

**Discussion by ss-torres** [microsoft/DTrace-on-Windows#50] [General]
I have the dtrace code shown below:

tick-10sec 
{
	exit(0);
}

struct ustr{uint16_t buffer[256];};

syscall::NtOpenFile:entry
/ execname == "Code.exe"/
{ 
	cidstr = (nt`_OBJECT_ATTRIBUTES*) (copyin(arg2, sizeof(nt`_OBJECT_ATTRIBUTES)));
	filename = (nt`_UNICODE_STRING*)copyin((uintptr_t)cidstr->ObjectName,sizeof(nt`_UNICODE_STRING));
	fname = (uint16_t*)
                copyin((uintptr_t)filename->Buffer,
                       filename->Length);
	printf("execname: %s, filename: %*ws", execname, filename->Length/2,
		((struct ustr*)fname)->buffer);
	stack(5);
	print("-----------------");
	ustack();
	print("==========");
}

END 
{
   printf("Printing stacks \n");
}

And the command printing are as following:

![image](https://github.com/microsoft/DTrace-on-Windows/assets/31719308/921fe768-4a8a-42c1-8ac6-084625f7c243)

We could see that the stack() function do nothing. So how could I get the kernel stack? And the user stack frame depth is small, and we could not get all what we want. Do you know how to get a deeper stack? Thank you.

**Comment by mhndr** [comment]
Hi , syscalls are just entry points to the kernel and not a lot of kernel functions have been invoked yet , therefore the call to stack is returning nothing for you. maybe if you tried to hook into some probe under "fbt:nt::entry" you might see a more deeper kernel stack. 

and regarding ustack not being too shallow my take would be that you need to hook into NtOpenFile syscall in the context of a specific process that you are trying to profile and also provide dtrace with the symbols of that process. doing that should give you the call stack that lead to the specific syscall that you're interested in
